analyzers/src/SonarAnalyzer.Common/Helpers/SymbolHelper.cs

```diff
@@ -209,6 +209,34 @@ public static bool IsPubliclyAccessible(this ISymbol symbol)
             symbol?.GetAttributes().Where(a => a.AttributeClass.IsAny(attributeTypes))
             ?? Enumerable.Empty<AttributeData>();
 
+        internal static IEnumerable<AttributeData> GetAttributesWithInherited(this ISymbol symbol)
+        {
+            foreach (var attribute in symbol.GetAttributes())
+            {
+                yield return attribute;
+            }
+
+            var baseSymbol = GetBaseSymbol(symbol);
+            while (baseSymbol != null)
+            {
+                foreach (var attribute in baseSymbol.GetAttributes().Where(x => x.AttributeUsageInherited()))
+                {
+                    yield return attribute;
+                }
+
+                baseSymbol = GetBaseSymbol(baseSymbol);
+            }
+
+            static ISymbol GetBaseSymbol(ISymbol symbol) =>
+                symbol switch
+                {
+                    INamedTypeSymbol namedType => namedType.BaseType,
+                    IMethodSymbol { OriginalDefinition: { } originalDefinition } method when !method.Equals(originalDefinition) => GetBaseSymbol(originalDefinition),
+                    IMethodSymbol { IsOverride: true, OverriddenMethod: { } overridenMethod } => overridenMethod,
```

Comment:
The `IsOverride` is redundant when `OverridenMethod` is null-checked
```suggestion
                    IMethodSymbol { OverriddenMethod: { } overridenMethod } => overridenMethod,
```